api/server/router/container/container_routes.go

```diff
@@ -769,12 +769,14 @@ func handleSysctlBC(
 			netIfSysctl := fmt.Sprintf("net.%s.%s.IFNAME.%s=%s", spl[1], spl[2], spl[4], v)
 			// Find the EndpointConfig to migrate settings to, if not already found.
 			if ep == nil {
+				/* TODO(robmry) - apply this to the API version used in 28.0.0
 				// Per-endpoint sysctls were introduced in API version 1.46. Migration is
 				// needed, but refuse to do it automatically for newer versions of the API.
-				if versions.GreaterThan(version, "1.46") {
```

Comment:
Looks like this already shipped, so maybe we https://github.com/moby/moby/blob/45d37a0ca928fedfa72eaede29a215eb2570b666/api/server/router/container/container_routes.go#L772-L777

Given that older daemon versions don't run "newer API versions", would it be cleaner to remove the code (for the 27.x release branch), and have a ("draft"?) "deprecation" PR to apply it? (that would also give a clear PR to point to "this PR deprecated / removed the feature")

Reply:
I'm probably missing the point, but I think ...
- this PR will need to be backported to the 27.x branch, for 27.2.0?
- we don't need to backport to a 27.1.x, as there won't be one and it wouldn't deal with API 1.47 anyway.

I raised issue https://github.com/moby/moby/issues/48282 as a reminder to re-apply the deprecation ... I don't think it can be a draft PR yet, because we don't know what the API version will be by the time master ships as 28.x?

Reply:
Sorry, I probably should've been more clear; so current master is targeting 28.0; we'll be doing a v27.2 release, but we may have to look if we need a v27.1.2 patch release before that; I'm at least aware of a regression in buildkit, which requires;

- https://github.com/moby/moby/pull/48246
- https://github.com/moby/moby/pull/48235

Given that we still want this code to be included in v28, we can either;

- Remove the code; backport to v27.x
- Open a PR to apply the code (which effectively is a "remove support for deprecated  ....")

Or

- Keep the code in master (but update the API version to `v1.47` together with the API version bump
- And open a PR specific for v27.x that removes the code altogether from that branch

Benefit of the last one is that we already run everything with that deprecation in place (so it would already be tested)

> I don't think it can be a draft PR yet, because we don't know what the API version will be by the time master ships as 28.x?

Yeah, that one is tricky; not _too_ problematic, but we must be sure that whenever we would do another API update to go out in a patch release before v28, we'd have to move the version up.

Reply:
FWIW; for a potential v27.1.2 release we need to discuss https://github.com/moby/sys/pull/140#issuecomment-2258876816

Reply:
I'd keep master/27.x in-step (with the deprecation code and its test commented out), to make merges less complicated. And depend on the issue against 28.0 to remember to un-comment the deprecation code and add the right version number in once we know what it is. The deprecation is trivial, not testing it until it's in-place and needed seems less likely to be an issue than forgetting to bump the number for each intermediate API version.

But, your call ... I'm happy to do go with any plan!